### ISSUE TITLE
Remote core is not working in openWRT

### ISSUE THREAD

**Issue by NguyenTrongThinh** [bootlin/openwrt-feed-st#2]
**I enable Linux configuration as below:**
CONFIG_REMOTEPROC=y
CONFIG_REMOTEPROC_CDEV=y
CONFIG_REMOTEPROC_SRM_CORE=y
CONFIG_REMOTEPROC_SRM_DEV=y
CONFIG_TEE_REMOTEPROC=y
CONFIG_STM32_RPROC=y
CONFIG_RPMSG=y
CONFIG_RPMSG_CHAR=y
CONFIG_RPMSG_CTRL=y
CONFIG_RPMSG_NS=y
CONFIG_RPMSG_TTY=y
CONFIG_RPMSG_VIRTIO=y
CONFIG_STM32_IPCC=y
**I checked the device tree, it looks ok.**
I got an error that the driver could not work
`[16.481009] platform 10000000.m4: deferred probe pending`
The remote core is not enable. What wrong with my configuration or this feature is not supported?
I use the branch master with latest commit. My hardware is STM32MP157F-DK2.
Thanks

**Comment by thom24** [comment]
In OpenWrt this feature is not supported.
When I integrated ST patches in the feed I had lots of conflicts, especially with the stm32_rproc driver.
For an initial support, the remoteproc feature was not a priority, so I skipped these patches.
So the stm32_rproc may not work.

I generated a patch to update the stm32_rproc to the version v6.1-stm32mp-r2: [965-update-stm32-rproc-to-v6.1-stm32mp-r2.patch.txt](https://github.com/user-attachments/files/18281733/965-update-stm32-rproc-to-v6.1-stm32mp-r2.patch.txt)
Sorry for the txt extension, I added it to be able to upload the patch.
You just have to add it in `target/linux/stm32/patches-6.1/`.
It is a bit aggressive, but I tested it and it works (remoteproc are available).
```
root@OpenWrt:~# ls /sys/class/remoteproc/remoteproc0/
coredump   device     fw_format  power      state      uevent
dev        firmware   name       recovery   subsystem
```

I will not commit this patch, as I'm working now on the integration of the BSP6.0.
